File: front/src/app/models/usuario.ts
export class Usuario {
  private _id: number;
  private _username: string;
  private _password: string;
  private _email: string;
  private _enabled: boolean;
  private _roles = [];

  constructor() {
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get username(): string {
    return this._username;
  }

  set username(value: string) {
    this._username = value;
  }

  get password(): string {
    return this._password;
  }

  set password(value: string) {
    this._password = value;
  }

  get enabled(): boolean {
    return this._enabled;
  }

  set enabled(value: boolean) {
    this._enabled = value;
  }

  get roles() {
    return this._roles;
  }

  set roles(value) {
    this._roles = value;
  }

  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }

  toJSON() {
    return {
      id: this.id,
      username: this.username,
      password: this.password,
      enabled: this.enabled,
      email: this.email,
      roles: this.roles
    };
  }
}

File: back/src/main/java/com/curso/spring/rest/model/entity/Usuario.java
package com.curso.spring.rest.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;
import javax.persistence.PrePersist;
import javax.persistence.Table;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collection;

/**
 * Entidad que representa a un usuario en la base de datos
 */
@Entity
@Table(schema = "tfg", name = "usuarios")
public class Usuario implements Serializable {

    private static final long serialVersionUID = 8091612874764696577L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true, length = 100)
    private String username;
    private String password;
    private Boolean enabled;

    // relación 1 a 1 con clientes
    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "cliente_id", referencedColumnName = "id")
    private Cliente cliente;

    // relación many to many con los roles
    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "usuarios_roles",
            joinColumns = @JoinColumn(
                    name = "user_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(
                    name = "rol_id", referencedColumnName = "id"))
    private Collection<Rol> roles;

    public Usuario() {
        this.roles = new ArrayList<>();
    }

    public Usuario(Cliente cliente, String password) {
        this();
        this.username = cliente.getEmail();
        this.password = password;
        this.setCliente(cliente);
    }

    public Integer getId() {
        return this.id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getUsername() {
        return this.username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return this.password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Boolean getEnabled() {
        return this.enabled;
    }

    public void setEnabled(Boolean enabled) {
        this.enabled = enabled;
    }

    public Collection<Rol> getRoles() {
        return this.roles;
    }

    public void setRoles(Collection<Rol> roles) {
        this.roles = roles;
    }

    public void addRol(Rol rol) {
        this.roles.add(rol);
    }

    @JsonIgnore
    public Cliente getCliente() {
        return this.cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    @PrePersist
    public void prePersist() {
        this.enabled = true;
    }
}
